### ISSUE TITLE
Not possible to use rst labels in embedded rst blocks with doxygen

### ISSUE THREAD

**Issue by t-b** [michaeljones/breathe#689]
Is it possibel to use rst labels in the embed:rst blocks?

I've tried to add that into the examples

```diff
diff --git a/examples/specific/rst.h b/examples/specific/rst.h
index 7c9a54b1..4d0c8ac0 100644
--- a/examples/specific/rst.h
+++ b/examples/specific/rst.h
@@ -11,9 +11,14 @@ public:

     This is some funky non-XML compliant text: <& !><

+    .. _mylabel1:
+
     .. note::

        This reStructuredText has been handled correctly.
+
+    See also :ref:`mylabel2`, :ref:`mylabel3` or :ref:`mylabel4`.
+
     \endrst

     This is just a standard verbatim block with code:
@@ -29,6 +34,9 @@ public:
     /*!
     Inserting additional reStructuredText information.
     \verbatim embed:rst
+
+    .. _mylabel2:
+
     .. note::

        This reStructuredText has been handled correctly.
@@ -40,6 +48,9 @@ public:
     * Inserting additional reStructuredText information.
     *
     * \verbatim embed:rst:leading-asterisk
+    *
+    * .. _mylabel3:
+    *
     *     Some example code::
     *
     *        int example(int x) {
@@ -56,6 +67,9 @@ public:
     /// @returns the same value provided in something param
     ///
     /// @verbatim embed:rst:leading-slashes
+    ///
+    /// .. _mylabel4:
+    ///
     ///    .. code-block:: c
     ///       :linenos:
     ///
--
```

See [0001-WIP-Add-labels-to-embedded-rst.zip](https://github.com/michaeljones/breathe/files/6411615/0001-WIP-Add-labels-to-embedded-rst.zip) for a patch file.

My use case is to link to specific parts of the embed:rst blocks from the rest of the rst documentation.

**Comment by vermeeren** [comment]
@t-b Not sure, does it work with the things you tested? Presumably not because you created the issue?

**Comment by t-b** [comment]
Ah okay, I should have mentioned that...

No I can't get it to work.

**Comment by vermeeren** [comment]
I am not familiar with this part of Breathe at all, it might have something to do with the way it injects the raw rst, possibly things are escaped in some way. One would expect that if raw rst is passed to Sphinx as-is this should work the way you are trying to use it.

**Comment by jakobandersen** [comment]
I don't think that is currently possible. The Doxygen XML is not parsed as RST as Breathe creates Docutils nodes directly. If there is a way to tag some text fragment in the Doxygen XML, then it should be possible to call the RST parser for that bit. I'm not too familiar with the Doxygen commands, but it could be that the [``\xmlonly``](https://www.doxygen.nl/manual/commands.html#cmdxmlonly) command may be useful.
Without having tried it, perhaps something like ``\xmlonly <breatheRST>.. myLabel:</breatheRST>`` would be enough on the user side, and Breathe then needs to learn how to visit such XML nodes.

**Comment by 2bndy5** [comment]
My initial approach would be to rely on doxygen more than breathe. By using the [`\anchor`](https://www.doxygen.nl/manual/commands.html#cmdanchor) cmd, you can then tell doxygen to `\ref` back to the declared anchor. Thus, breathe can do what it was originally designed to do: translate doxygen. I haven't tested this suggestion though, so I'm not sure if the `\anchor` cmd is supported by breathe.

**Comment by arwedus** [comment]
> I don't think that is currently possible. The Doxygen XML is not parsed as RST as Breathe creates Docutils nodes directly. If there is a way to tag some text fragment in the Doxygen XML, then it should be possible to call the RST parser for that bit. I'm not too familiar with the Doxygen commands, but it could be that the [`\xmlonly`](https://www.doxygen.nl/manual/commands.html#cmdxmlonly) command may be useful. Without having tried it, perhaps something like `\xmlonly <breatheRST>.. _myLabel:</breatheRST>` would be enough on the user side, and Breathe then needs to learn how to visit such XML nodes.

I can't understand how that would work for the `embed:rst` sections. They need to be parsed as rST, right?

**Comment by jakobandersen** [comment]
> > I don't think that is currently possible. The Doxygen XML is not parsed as RST as Breathe creates Docutils nodes directly. If there is a way to tag some text fragment in the Doxygen XML, then it should be possible to call the RST parser for that bit. I'm not too familiar with the Doxygen commands, but it could be that the [`\xmlonly`](https://www.doxygen.nl/manual/commands.html#cmdxmlonly) command may be useful. Without having tried it, perhaps something like `\xmlonly <breatheRST>.. _myLabel:</breatheRST>` would be enough on the user side, and Breathe then needs to learn how to visit such XML nodes.
> 
> I can't understand how that would work for the `embed:rst` sections. They need to be parsed as rST, right?

Right, my old comment is not correct. It looks like the data is indeed passed through the docutils, in the method [``visit_verbatim``](https://github.com/michaeljones/breathe/blob/26844b77d805a93bd0df3e14a289d23adcf8ab19/breathe/renderer/sphinxrenderer.py#L1642), after some potential transformation.

It would be nice if one of you could provide an archive with  small stand-alone project that illustrates the problem.

**Comment by t-b** [comment]
> It would be nice if one of you could provide an archive with small stand-alone project that illustrates the problem.

@jakobandersen https://github.com/t-b/breathe/tree/feature/embedded-labels-rst has the code from the first post here applied to breathe.

**Comment by jakobandersen** [comment]
> > It would be nice if one of you could provide an archive with small stand-alone project that illustrates the problem.
> 
> @jakobandersen https://github.com/t-b/breathe/tree/feature/embedded-labels-rst has the code from the first post here applied to breathe.

Well, I wouldn't consider Breathe it self a small stand-alone project :-), but more a minimum example with a single hpp and rst file, so we are clear about what the problem is and how to reproduce it. Otherwise I will have to look into it at some point when I have more time.
